chia/data_layer/s3_plugin_service.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import asyncio
+import concurrent.futures
+import functools
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Dict, List
+from urllib.parse import urlparse
+
+import boto3 as boto3
+import yaml
+from aiohttp import web
+from botocore.exceptions import ClientError
+
+from chia.types.blockchain_format.sized_bytes import bytes32
+
+log = logging.getLogger(__name__)
+
+
+class S3Plugin:
+    boto_client: boto3.client
+    port: int
+    region: str
+    aws_access_key_id: str
+    aws_secret_access_key: str
+    store_ids: List[str]
+    bukets: Dict[str, List[str]]
+    urls: List[str]
+    instance_name: str
+
+    def __init__(
+        self,
+        region: str,
+        aws_access_key_id: str,
+        aws_secret_access_key: str,
+        store_ids: List[str],
+        bukets: Dict[str, List[str]],
+        urls: List[str],
+        instance_name: str,
+    ):
+        self.boto_client = boto3.client(
+            "s3",
+            region_name=region,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        self.store_ids = store_ids
+        self.bukets = bukets
+        self.urls = urls
+        self.instance_name = instance_name
+
+    async def check_store_id(self, request: web.Request) -> web.Response:
+        self.update_instance_from_config()
+        try:
+            data = await request.json()
+        except Exception as e:
+            print(f"failed parsing request {request} {e}")
+            return web.json_response({"handles_url": False})
+        store_id = bytes32.from_hexstr(data["id"])
+        if store_id.hex() in self.store_ids:
```

Comment:
I would rather `bytes32.from_hexstr()` the config data than `.hex()` this.  Make all the in-memory operations use non-serialized forms.

Reply:
what do you mean by "in mempry operations ?
i dont want to not use the conversions so we can throw in case we gat invalid input

Reply:
A hexadecimal encoded string of bytes is a serialization of the data.  I think this should be used when serialization over a network connection or to a file is needed (and in this case that those mechanisms are using something like JSON that can't handle bytes directly).  Other things like storage in dicts and comparisons etc would be "in memory operations".

I like that the request `data["id"]` is processed as you have it, `store_id = bytes32.from_hexstr(data["id"])`.  I am suggesting that the `self.store_ids` be populated in the same way instead of storing the hexadecimal serialization.  Then we would just have `if store_id in self.store_ids:`

I probably should have raised this issue at the code populating `self.store_ids` or such.